### PR TITLE
Librarians now start with one additional language

### PR DESCRIPTION
**Pull request by JustSumBody**
For obvious metagay reasons, restricted languages like mouse and cult are not included in the possible languages.
:cl:
 * rscadd: Librarians now start knowing one additional random language.

### CODE REVIEW

**Comment by 9600bauds** [comment]
OK, better idea:
Random language books spawn in the middle of the 6000 doujins, you must spot the odd title out and read it to get your languages

**Comment by Probe1** [comment]
Anything to flesh out librarians ok

File: code/game/jobs/job/civilian.dm [hunk 1]
```diff
@@ -421,6 +421,15 @@
 			H.put_in_hand(GRASP_RIGHT_HAND, new H.species.survival_gear(H))
 		else
 			H.equip_or_collect(new H.species.survival_gear(H.back), slot_in_backpack)
+		var/list/new_languages = list()
+		for(var/L in all_languages)
+			var/datum/language/lang = all_languages[L]
+			if(!(lang.flags & RESTRICTED) && !(lang in H.languages))
```
Comment (PJB3005) [inline code/game/jobs/job/civilian.dm]: `~lang.flags & RESTRICTED` instead.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Review by PJB3005 — APPROVED** [review]

File: code/game/jobs/job/civilian.dm [hunk 2]
```diff
@@ -421,6 +421,15 @@
...
 			H.equip_or_collect(new H.species.survival_gear(H.back), slot_in_backpack)
+		var/list/new_languages = list()
+		for(var/L in all_languages)
+			var/datum/language/lang = all_languages[L]
+			if(~lang.flags & RESTRICTED && !(lang in H.languages))
+				new_languages += lang.name
+
+		var/picked_lang = pick(new_languages)
+		H.add_language(picked_lang)
+		to_chat(H, "Due to your well read nature, you find yourself versed in the language of [picked_lang].")
```
Comment (ComicIronic) [inline code/game/jobs/job/civilian.dm]: Better add how to use the language in question, for a new player.

Comment (kilozombie) [reply code/game/jobs/job/civilian.dm]: Appears in 'known languages', not sure if that's really necessary.

**Comment by ghost** [comment]
Does this check if you would already know the language?

**Comment by PJB3005** [comment]
@Skullyton yes.

File: code/game/jobs/job/civilian.dm [hunk 3]
```diff
@@ -421,6 +421,15 @@
 			H.put_in_hand(GRASP_RIGHT_HAND, new H.species.survival_gear(H))
 		else
 			H.equip_or_collect(new H.species.survival_gear(H.back), slot_in_backpack)
+		var/list/new_languages = list()
+		for(var/L in all_languages)
+			var/datum/language/lang = all_languages[L]
+			if(~lang.flags & RESTRICTED && !(lang in H.languages))
```
Comment (gbasood) [inline code/game/jobs/job/civilian.dm]: I don't think `~` does what you think it does here

Comment (gbasood) [inline code/game/jobs/job/civilian.dm]: I'm wrong though

**Review by gbasood — APPROVED** [review]

**Comment by murnandreams** [comment]
But why can't we be the mice master librarian?

**Comment by AI-Is-Rogue** [comment]
Mouse king sawbones, the librarian

**Comment by ihadtoregisterforthis** [comment]
**squeek squeek, it's nukeops, they want to enter by the singularity, kill me so I can watch them and I'll be back in 5 minutes, admins are AFK, squeek\*\*

**Comment by murnandreams** [comment]
Well, they'll just eat a bannu for ock ick. Same concept as diona joining with a meta name, or ghosts pointing chairs at antags.

**Pull request merged**